certbot/log.py

```diff
@@ -299,7 +300,23 @@ def except_hook(exc_type, exc_value, trace, debug, log_path):
             _, _, exc_str = str(exc_value).partition(':: ')
             print(exc_str, file=sys.stderr)
         else:
-            traceback.print_exception(exc_type, exc_value, None)
+            """ Regex for parsing error messages"""
+            """ Regular expression groups"""
+            """ Match Group #1: Attempted connection host"""
+            """ Match Group #2: urllib3 Error Number"""
+            """ Match Group #3: urllib3 Error String"""
+            """ Example:"""
+            """ Match Group #1: acme-v01.api.letsencrypt.org"""
+            """ Match Group #2: [Errno 65]"""
+            """ Match Group #3: No route to host"""
+            err_re = re.search(r"host='(\S*)'.*(\[Errno \d+\]\ )(.*)(',\)\))", str(exc_value))
+            if err_re:
+                err_msg = "Can't connect to "+err_re.group(1)+\
```

Comment:
@jsha - what parts of the message do you think are important to capture?  I want to make sure that the error messaging is sensible to the person reading the error message.  I'd propose:

hostname
[urllib error code] - urllib error message


So we would net an output something like so:

```
Saving debug log to /var/log/letsencrypt/letsencrypt.log

An unexpected error occurred:
Can't connect to acme-v01.api.letsencrypt.org
[Errno -2] Name or service not known

Please see the logfiles in /var/log/letsencrypt for more details.
```